### PR TITLE
fix(nuxt): remove modulepreload for spa fallback routes with `ssr:true`

### LINKED ISSUE #7548: generated 200.html references /200.html/_payload.js
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.10`
- Nitro Version:    `0.5.3`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

- Run `nuxt generate` on nuxt3 project.

### Describe the bug

It's great to have the fallback file `200.html` now! 💖

However, the fallback file `200.html` references a `/200.html/_payload.js` which does not exist.

```html
<!DOCTYPE html>
<html >
<head><link rel="modulepreload" href="/200.html/_payload.js"><link rel="modulepreload" as="script" crossorigin...
```

This causes this error in the console as the `200.html` is served as a fallback 😁
```
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
```

Not sure if this is the reason, the fallback is not working properly, but let's take one step at a time 😉

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by pi0**
### 🔗 Linked issue

closes #7548

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

[7548](https://github.com/nuxt/framework/issues/7548#issuecomment-1248003267)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/core/runtime/nitro/renderer.ts [hunk 1]
```diff
@@ -194,7 +194,7 @@ export default defineRenderHandler(async (event) => {
     htmlAttrs: normalizeChunks([renderedMeta.htmlAttrs]),
     head: normalizeChunks([
       renderedMeta.headTags,
-      !process.env.NUXT_NO_SCRIPTS && process.env.prerender ? `<link rel="modulepreload" href="${payloadURL}">` : null,
+      (process.env.prerender && !ssrContext.noSSR) ? `<link rel="modulepreload" href="${payloadURL}">` : null,
```
Comment (danielroe) [inline packages/nuxt/src/core/runtime/nitro/renderer.ts:197]:
```suggestion
      (process.env.prerender && !ssrContext.noSSR && !process.env.NUXT_NO_SCRIPTS) ? `<link rel="modulepreload" href="${payloadURL}">` : null,
```

Comment (danielroe) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:197]: But this way we can treeshake it.

Comment (danielroe) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:197]: If we want to have the ability to switch it on and off via ssrContext.noSSR then we should also update in L213-218.

Comment (pi0) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:197]: There is no tree-shaking benefit. `env.prerender` is already causing it.

Comment (danielroe) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:197]: There is a tree shaking benefit if we are not prerendering.

Comment (pi0) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:197]: Have to tried building to confirm?

image

**Pull request merged**